src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs

```diff
@@ -429,19 +433,21 @@ internal Location CreateLocation(AnnotatedCodeLocationVersionOne v1AnnotatedCode
                 Message = CreateMessage(message)
             };
 
-            LogicalLocation logicalLocation = new LogicalLocation();
+            logicalLocationKey = logicalLocationKey ?? fullyQualifiedLogicalName;
```

Comment:
>logicalLocationKey [](start = 12, length = 18)

Hold on there, bucko :-)

If FQLN is present but key is absent, then the lookup on Line 440 isn't guaranteed to succeed. `run.logicalLocation` might not even exist. The `logicalLocation` will be `null`, and from Line 442 on it will all end in tears.

You have to check the result of the lookup, and if it fails, synthesize your own `LogicalLocation` object as you did on Line 432 of the left-hand side.